hack/run-molecule-tests.sh

```diff
@@ -354,6 +354,12 @@ do
     printf '%s [%s]' "$(green 'success')" "${duration}"
   else
     printf '%s [%s]' "$(red 'FAILURE')" "${duration}"
+
+    # If we are in CI, abort fast. Also, print logs of the failed test to allow for inspection
+    if [ "${CI}" == "true" ]; then
+      cat ${TEST_LOGS_DIR}/${t}.log
+      exit 1
+    fi
```

Comment:
This is no good - because the "exit 1" is going to exit the entire script so it won't continue to run the other molecule tests. So if the very first molecule test fails, the entire rest of the molecule test suites never run. We don't want that. We want all of them to run so we can see which molecule tests fail and which succeed.

This is why we already have the ability to store the logs in a logs repo. In this script you already have this feature - you just have to enable it and prepare a logs repo. See:

* https://github.com/kiali/kiali/blob/master/hack/ci-kind-molecule-tests.sh#L87-L93
* https://github.com/kiali/kiali/blob/master/hack/ci-kind-molecule-tests.sh#L123-L128

You can pair this together with this other script I have so we retain the last N sets of molecule test logs in the logs repo: https://github.com/jmazzitelli/mazz-basement/blob/master/server-scripts/purge-directories-from-git.sh

I use this on mazz-basement by running the openshift molecule tests like this:

```
/home/jmazzite/source/kiali/kiali/src/github.com/kiali/kiali/hack/ci-openshift-molecule-tests.sh -dorp podman -ir kiali -ul true > /tmp/KIALI-TESTS.log 2>&1 && /home/jmazzite/source/mazz-basement/server-scripts/purge-directories-from-git.sh --do-not-push false >> /tmp/KIALI-TESTS.log 2>&1
```


The result in the logs repo will look like this (this is the mazz-basement test results - notice there is only ever the last 10 nightly runs that are retained): https://github.com/jmazzitelli/kiali-molecule-test-logs

You can see all the directories there have a "results.log" - that shows you all the molecule tests and their success/failure (and this is why your update here isn't good - as soon as one of those tests results in FAILURE, no other tests will run after that).